CMakeLists.txt

```diff
@@ -72,7 +72,6 @@ option(INSTALL_DOC "Build/install documentation" ON)
 option(INSTALL_LIB "Install the library" OFF)
 tr_auto_option(ENABLE_DEPRECATED "Allow deprecated API use of upstream packages, e.g. GTK" AUTO)
 tr_auto_option(RUN_CLANG_TIDY "Run clang-tidy on the code" OFF)
-tr_auto_option(USE_SYSTEM_EVENT2 "Use system event2 library" AUTO)
```

Comment:
I'd like to still support this option, some package maintainers will not like it being removed. If system libevent is built w/o openssl support then SSL support in RPC will be disabled.

Reply:
Revert this option but set it to `OFF` by default. And if use system libevent then `libevent_openssl` will be disabled.

Reply:
I know you mean well but I fear your motivation here could be wrong. If you want it to default to `OFF` because `AUTO`/`ON` will mean lack of SSL support then the right thing to do is adjust cmake/FindEVENT2.cmake so that it actually looks for libevent_openssl on the system so that SSL support is possible.

Since silently enabling/disabling a feature based on search results or other features is usually wrong (it leads to unpredictable build results), I'd say we also want SSL support in RPC being controlled by a separate option which only becomes available if openssl backend is selected (and defaults to `AUTO` in this case), but which allows users to explicitly enable/disable the feature if they want to make sure it's enabled/disabled (i.e. leading to a configuration error, not a silent disabling, if such support can't be enabled). I believe wanting to use openssl backend for crypto operations but not wanting SSL support in RPC is a valid option; at the same time, I understand that it might feel like an overcomplication for no substantial gain, so I'd like for @ckerr or other @transmission/core-devs to chime in here.

Reply:
Adjusted `cmake/FindEVENT2.cmake` to find `libevent_openssl`